app/components/UI/Modal.js

```diff
@@ -28,6 +28,27 @@ class Modal extends React.Component {
     withHeader: false
   }
 
+  constructor(props) {
+    super(props)
+    this.closeOnEscape = this.closeOnEscape.bind(this)
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.closeOnEscape)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.closeOnEscape)
+  }
+
+  closeOnEscape(e) {
+    if (e.key === 'Escape') {
+      const { onClose } = this.props
+
+      onClose()
```

Comment:
`onClose` is not a required prop, and there are places where we display the modal without this prop set (such as in the onboarding process for brand new users). We need to ensure it's defined here before trying to call it.

Reply:
Will do.